util/project_allocation_mapper.py

```diff
@@ -91,7 +91,18 @@ def get_all_projects(self):
         else:
             for tas_project in self._tas_all_projects():
                 projects[tas_project['chargeCode']] = tas_project
-        return projects.values()
+        values = projects.values()
+        values.sort(reverse=True,key=self.sort_by_allocation_request_date)
+        return values
+
+    '''
+    Sort by most recent allocation request
+    '''
+    def sort_by_allocation_request_date(self, el):
+        if el['allocations']:
+            return datetime.strptime(el['allocations'][0]['dateRequested'], '%Y-%m-%dT%H:%M:%SZ')
```

Comment:
Is the allocation in position 0 always going to be the most recent? It might make sense to get the max `dateRequested`.

Also, what if `dateRequested` is not on the allocation? I think this would throw an error in that case, which can probably break a lot of pages. Perhaps defaulting to datetime.min and logging an error is best.

Reply:
good catch, yes it looks like the last item in the allocations list is the most recent. I think you're right on `dateRequested` too, if it happens to be missing we'll probably get an error, i'll make updates and resubmit, thanks!